src/pages/ReimbursementAccount/ReimbursementAccountPage.js

```diff
@@ -97,116 +97,175 @@ const defaultProps = {
-                // this.props.reimbursementAccount.isLoading just changed to false. From now on, it makes sense to run the code
-                // below updating states and the route, and this will happen in the next react lifecycle.
-                this.setState({
-                    shouldShowContinueSetupButton: this.getShouldShowContinueSetupButtonInitialValue(),
-                    hasACHDataBeenLoaded: true,
-                });
+    });
+    const [shouldShowContinueSetupButton, setShouldShowContinueSetupButton] = useState(
+        hasACHDataBeenLoaded ? getShouldShowContinueSetupButtonInitialValue() : false
+    );
+    const [hasACHDataBeenLoaded, setHasACHDataBeenLoaded] = useState(reimbursementAccount !== ReimbursementAccountProps.reimbursementAccountDefaultProps);
+
+    const achData = lodashGet(reimbursementAccount, 'achData', {});
+    const currentStep = achData.currentStep || CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT;
+    const policyName = lodashGet(policy, 'name');
+    const policyID = lodashGet(route.params, 'policyID');
+
+    const continueFunction = () => {
+        setShouldShowContinueSetupButton(false);
+        fetchData(true);
+    };
+
+    useEffect(() => {
+        // If it's a subsequent render, compare the current props with prevProps if needed.
+        // At the end of the useEffect, update the prevProps to the current props for the next render cycle.
+        setPrevProps({
+            current: {
+                reimbursementAccount,
+                network,
+                route,
+                onfidoToken,
+                policy,
+                account,
+                isLoadingReportData,
+                session,
+                translate,
+                plaidLinkToken,
+                reimbursementAccountDraft,
             }
-            return;
-        }
+        });
+    }, [
+        reimbursementAccount,
+        network,
+        route,
+        onfidoToken,
+        policy,
+        account,
+        isLoadingReportData,
+        session,
+        translate,
+        plaidLinkToken,
+        reimbursementAccountDraft,
+    ]);
 
-        if (
-            prevProps.reimbursementAccount.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE &&
-            this.props.reimbursementAccount.pendingAction !== prevProps.reimbursementAccount.pendingAction
-        ) {
-            // We are here after the user tried to delete the bank account. We will want to set
-            // this.state.shouldShowContinueSetupButton to `false` if the bank account was deleted.
-            this.setState({shouldShowContinueSetupButton: this.hasInProgressVBBA()});
+    /**
+    * @param {String} fieldName
+    * @param {*} defaultValue
+    *
+    * @returns {*}
+    */
+    const getDefaultStateForField = (fieldName, defaultValue = '') => {
+        return lodashGet(reimbursementAccount, ['achData', fieldName], defaultValue);
+    };
+
+    const goBack = () => {
```

Comment:
Let's add empty lines between different code blocks. It looks a little bit complex for reading.

Reply:
I agree, I would add a lot of such styling spacings but been said no to add them :) I added space before `switch` as I indeed missed it. Thank you.